web/modules/custom/dpl_favorites_list_material_component/src/Plugin/Block/FavoritesListMaterialComponentBlock.php

```diff
@@ -0,0 +1,100 @@
+
+  /**
+   * FavoritesListMaterialComponentBlock constructor.
+   *
+   * @param mixed[] $configuration
+   *   A configuration array containing information about the plugin instance.
+   * @param string $plugin_id
+   *   The plugin ID for the plugin instance.
+   * @param mixed $plugin_definition
+   *   The plugin implementation definition.
+   * @param \Drupal\Core\Config\ConfigFactoryInterface $configFactory
+   *   Drupal config factory to get FBS and Publizon settings.
+   */
+  public function __construct(array $configuration, $plugin_id, $plugin_definition, ConfigFactoryInterface $configFactory) {
+    parent::__construct($configuration, $plugin_id, $plugin_definition);
+    $this->configuration = $configuration;
+    $this->configFactory = $configFactory;
+  }
+
+  /**
+   * {@inheritDoc}
+   *
+   * @param \Symfony\Component\DependencyInjection\ContainerInterface $container
+   *   The service container.
+   * @param mixed[] $configuration
+   *   A configuration array containing information about the plugin instance.
+   * @param string $plugin_id
+   *   The plugin ID for the plugin instance.
+   * @param int $plugin_definition
+   *   The plugin implementation definition.
+   *
+   * @return \Drupal\dpl_favorites_list_material_component\Plugin\Block\FavoritesListMaterialComponentBlock|static
+   *   Favorites list material component block.
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
+    return new static(
+      $configuration,
+      $plugin_id,
+      $plugin_definition,
+      $container->get('config.factory'),
+    );
+  }
+
+  /**
+   * {@inheritDoc}
+   *
+   * @return mixed[]
+   *   The app render array.
+   */
+  public function build() {
+    $favoritesListMaterialComponentSettings = $this->configFactory->get('favorites_list_material_component.settings');
+
+    $data = [
+      // Urls.
+      'favorites-list-material-component-go-to-list-url' => $favoritesListMaterialComponentSettings->get('favorites_list_url'),
+      'dpl-cms-base-url' => DplReactAppsController::dplCmsBaseUrl(),
+      // Texts.
+      'favorites-list-material-component-title-text' => $this->t("Your list", [], ['context' => 'Favorites list material component']),
+      'material-by-author-text' => $this->t("By", [], ['context' => 'Favorites list material component']),
+      'material-and-author-text' => $this->t("and", [], ['context' => 'Favorites list material component']),
+      'favorites-list-material-component-go-to-list-text' => $this->t("Go to My list", [], ['context' => 'Favorites list material component']),
+      'add-to-favorites-aria-label-text' => $this->t("Add element to favorites list", [], ['context' => 'Favorites list material component (aria)']),
+      'remove-from-favorites-aria-label-text' => $this->t("Remove element from favorites list", [], ['context' => 'Favorites list material component (aria)']),
```

Comment:
Consider sorting the lines. Better for maintainability

Reply:
Fixed